bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -304,6 +304,36 @@ public void resumeMinorGC() {
         }
     }
 
+
+    /**
+     * gc ledger storage.
+     */
+    void gc() {
+        LOG.info("Starting garbage collecting ledger storage.");
+
+        LOG.info("Garbage collecting deleted ledgers' index files.");
+        // gc inactive/deleted ledgers
+        doGcLedgers();
+
+        LOG.info("Scanning entry log files to extract metadata and delete empty entry logs if possible.");
+        // Extract all of the ledger ID's that comprise all of the entry logs
+        // (except for the current new one which is still being written to).
+        entryLogMetaMap = extractMetaFromEntryLogs(entryLogMetaMap);
+
+        // if it isn't running, break to not access zookeeper
+        if (!running) {
+            return;
+        }
+
+        LOG.info("Garbage collecting deleted ledgers' index files again just in case ledgers deleted during scanning.");
+        // gc inactive/deleted ledgers again, just in case ledgers are deleted during scanning entry logs
+        doGcLedgers();
```

Comment:
if the scan takes too longer time, it makes sense to scan the ledgers again to figure out what ledgers might have been deleted. I added a timer to explicitly check the scan time.